stingray/covariancespectrum.py

```diff
@@ -11,27 +12,43 @@
 
 class Covariancespectrum(object):
 
```

Comment:
yes, it's super confusing, but I don't know how to simplify it except request people to just always pre-make their light curves and reference light curves, or require input as events only.

I could add a couple of examples to the docstring? Or we could make sure the tutorial is really detailed (which is going to need updating anyway)?

Reply:
Let's work on the latter. The changes look good! It's overall more powerful than before and also avoids some duplication. 

I don't know any test examples where we know the covariance, besides the clear case of the same light curve in both channels. If we can add some better test on this it would be great, but for me, it's already mergeable.